### PR TITLE
Navigating resources filter

### LINKED ISSUE #2369: [Resources] Navigation in resources
**Current Behavior**
Currently, the Resources area looks exactly like the Blog area

**Desired Situation**
We're in need of figuring out how we want to change the way visitors navigate this interface so that it isn't just a repeat of the blog collection, but a perusable catalog of topics. It's ok to include repetitive content from /events, /blogs, /news. Filters that we'll want to account for on this page include content:

```
TYPE

- Case Study
- Presentation
- Demo
- Recorded Webinar
- Interview
- Blog
- News
- White Paper
- Article
- Tutorial
- FAQ
- Guide
- Podcast

---------------------

PRODUCT

- Meshery
- Service Mesh Performance
- Nighthawk

---------------------

TECHNOLOGY

- Cloud
- Kubernetes
- Docker
- WebAssembly
- JWT
- API

---------------------

SERVICE MESH

- Consul
- Linkerd
- Istio
- Open Service Mesh
- Kuma
- Network Service Mesh
- Traefik Mesh

```

For inspiration see:
1.  https://www.hashicorp.com/resources

The way that the categories and filters are on HashiCorp's resources page is perfect. The categories colorize onHover. You can clear the filter. It's great - https://www.hashicorp.com/resources?products=terraform&products=consul

See other examples:
2.  https://datadoghq.com/resources 
3.  https://d2iq.com/resources

---
**Contributor Resources**

The layer5.io website uses Gatsby, React, and GitHub Pages. Site content is found under the [`master` branch](https://github.com/layer5io/layer5/tree/master).
- See [contributing instructions](https://github.com/layer5io/layer5/blob/master/CONTRIBUTING.md)
- See Layer5 site designs in this [Figma project](https://www.figma.com/file/5ZwEkSJwUPitURD59YHMEN/Layer5-Designs). Join the [Layer5 Community](http://slack.layer5.io) for access.

### PR DESCRIPTION
**Pull request by debo19**
**Description**
Added filters to the Resources Page for easier navigation.

This PR fixes #2369

**Notes for Reviewers**
1. This only filters the items under the "Resource" collection. I'm yet to incorporate blogs and other collection types into the resource library.
2. I've added test properties to some of the articles in the collection to check out the functionality. We need to go through all articles in the collection and add appropriate `type`, `product`, `technology`, and `mesh`.

**[Signed commits](https://github.com/layer5io/layer5/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by l5io** [comment]
🚀 Preview for commit c8e4186a21062bc252a26fc40d765731e5c50605 at: https://61c63e68338fb07ef35ada27--layer5ng.netlify.app

**Comment by warunicorn19** [comment]
@debo19 just checked it out, looking really good. Just a question though is the approach an intersection or union? Also, a small nit 😅 the checkboxes aren't aligned with the labels.

File: src/sections/Resources/Resources-grid/filters.js [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+    const [expandType, setExpandType] = useState(true);
+    const [expandProduct, setExpandProduct] = useState(false);
+    const [expandTech, setExpandTech] = useState(false);
+    const [expandMesh, setExpandMesh] = useState(false);
+
+    const data = React.useMemo(() => options);
+    let typeOptions = data.filter((data) => data.category === "Type");
+    let productOptions = data.filter((data) => data.category === "Product");
+    let techOptions = data.filter((data) => data.category === "Technology");
+    let meshOptions = data.filter((data) => data.category === "Service Mesh");
```
Comment (warunicorn19) [inline src/sections/Resources/Resources-grid/filters.js:17]: If I'm not wrong this portion here is not required right? We can get this from the ResourceList.js file?? I might be wrong though

**Review by warunicorn19 — COMMENTED** [review]
Code-wise everything looks good.

File: src/sections/Resources/Resources-grid/filters.js (hunk 1 above)
Comment (debo19) [reply src/sections/Resources/Resources-grid/filters.js:17]: This is required but I need to find a better way to do it. I was mapping over the options file just once but that was messing with the collapsible so I left it this way for now. I'll look into it though.

**Comment by debo19** [comment]
> @debo19 just checked it out, looking really good. Just a question though is the approach an intersection or union? Also, a small nit 😅 the checkboxes aren't aligned with the labels.

@warunicorn19 So the selected options within each category have a union. The collection of all the options in each category (`type`, `product`, `technology`, `service mesh`) has an intersection. I hope I made sense 😅

**Comment by warunicorn19** [comment]
@debo19 if you could please fix the alignment, then we can go ahead and merge this, we can work on refactoring the code later on.

**Comment by l5io** [comment]
🚀 Preview for commit c5252b29bbb180eb205d244f58f9090bd3e9bc91 at: https://61c788f642fc11286cb5a8de--layer5ng.netlify.app

**Comment by leecalcote** [comment]
Getting closer...

**Comment by l5io** [comment]
🚀 Preview for commit facfadefdad1b249e827f7a39501b1e14d5ad852 at: https://61cc80848b3f27d5e50aad8f--layer5ng.netlify.app

**Comment by leecalcote** [comment]
@debo19 I love how you give full consideration in your work. 

- ability to clear filters
- ability to multiple filters
- show an empty state message
- soft transition on display of new filters categories
- hide unused categories so as to not overload user

Enhancement requests: 

- [ ] a small label on each card that identifies the resource type (see https://meshery.io/service-mesh-catalog for inspiration).
- [ ] a total number of resources available in each category with a small "(##)" behind the filter option; e.g. "Article (12)"
- [ ] make the filter category title clickable in addition to the chevron
- [ ] change empty state verbiage from "No resources match the applied filters" to "No matching resources" (leaving the sub-message intact).

**Comment by l5io** [comment]
🚀 Preview for commit 6ab56ed7003eb10ddbecd0e15d027fe89f461302 at: https://61cdbc286f54ff8c372aa99f--layer5ng.netlify.app

**Comment by debo19** [comment]
@leecalcote Since we don't have resources associated with every filter at the moment, not all of them show a number beside the categories. I'm looking for a way to add "(0)" but I'm not sure how that will look. I've made the other enhancements.

**Comment by warunicorn19** [comment]
@debo19 could you please hide the filters menu or keep it completely collapsed in the mobile view. Right now it covers up a lot of space in mobile view. Though, we can probably work on that separately.

**Review by warunicorn19 — APPROVED** [review]
LGTM! good stuff @debo19, @leecalcote should we move forward with merging this now?

**Comment by leecalcote** [comment]
Sounds real good! A separate issue for the mobile view item....

**Pull request merged**